src/utils/mashupDocumentParser.ts

```diff
@@ -150,3 +161,131 @@ export const editSingleQueryMetadata = (metadataArray: Uint8Array, metadata: Met
 
     return newMetadataArray;
 };
+
+const addConnectionOnlyQueryMetadata = (metadataArray: Uint8Array, connectionOnlyQueryNames: string[]) => {
+        // extract metadataXml
+        const mashupArray: ArrayReader = new arrayUtils.ArrayReader(metadataArray.buffer);
```

Comment:
nit: can you put it in a different method extractAndParseMetadataXml and call it from editSingleQueryMetadata also?